[Docs] Update "Alert and action settings" docs to be generated from YAML source

kilfoyle opened the pull request:
This PR:
 - Updates the Kibana [Alert and action settings](https://www.elastic.co/guide/en/kibana/current/alert-action-settings-kb.html) page to be based off of a YAML source file (`/docs/settings-gen/source/kibana-alert-action-settings.yml`) that is manually converted to Asciidoc format (`kibana-alert-action-settings.asciidoc`) by means of a Perl script (`docs/settings-gen/parse-settings.pl`). A preview of the new, generated page is [here](https://kibana_bk_191787.docs-preview.app.elstc.co/guide/en/kibana/master/alert-action-settings-kb.html).
 - Adds the `docs/settings-gen/parse-settings.pl` script which does the YAML → Asciidoc conversion.

All new files are added to the `/docs/source-gen` folder. 

This is a trial run updating only one page of settings in the docs. Later, in separate PRs, we plan to convert other pages. After all Kibana settings pages have been converted, we would ask that the Perl script be run automatically as part of the CI whenever the YAML files in `/docs/source-gen` are added or updated.

**Notes:**
 - The Docs team is happy to own and maintain the Perl script (sorry to use Perl - it's the only scripting language that I know). 
 - In time we also plan to convert all of these files from Asciidoc to Markdown.
 - When we eventually/hopefully get the rest of the Kibana settings files converted, we will announce the settings doc process to the Kibana team by email and/or in the Kibana newsletter.

Big thanks to the amazing @lukeelmers and @KOTungseth for guiding this!

---
Why are we doing this? We aim to:
 - Create a more consistent appearance for settings across all of the docs.
 - Make it easier for people to contribute, since all Asciidoc/Markdown formatting is handled by a script.
 - Make it more apparent which settings may be missing info, such as the default values, available options, etc.
---

P.S. I haven't worked in the Kibana repo very much and would appreciate any help navigating the CI checks. 

Rel: https://github.com/elastic/docs-projects/issues/239

elasticmachine commented:
Pinging @elastic/kibana-docs (Team:Docs)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241164)
* Commit: 5c31ac4b69242527767355b204c36dca57471f79

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/241164#0192769e-72fa-4881-862d-9bc94dd0ccbb)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241164","number":241164,"commit":"5c31ac4b69242527767355b204c36dca57471f79"}],"number":241164}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241218)
* Commit: df17b448c949ea11b8207607de57a62a165c5efc

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/241218#019276fc-976d-439e-b341-13a99c6172a7)
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/241218#01927727-07d9-4a2d-96ca-91b68676b6c0)

### History
* :broken_heart: [Build #241164](https://buildkite.com/elastic/kibana-pull-request/builds/241164) failed 5c31ac4b69242527767355b204c36dca57471f79

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241218","number":241218,"commit":"df17b448c949ea11b8207607de57a62a165c5efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241164","number":241164,"commit":"5c31ac4b69242527767355b204c36dca57471f79"}],"number":241218}
buildkite-pr-comment-->

lukeelmers reviewed (commented):
The overall approach here makes sense to me (dev team maintains `yml` files, docs team maintains script to convert `yml` to `asciidoc`).

A few things I have questions about (and pinging the relevant teams to provide input):
- Most important thing to review: are we generally comfortable with the yml schema design here? Because this will dictate the authoring experience for yml config docs in the future. @elastic/kibana-operations @kobelb & others
- Is the current directory the right location for the conversion script or does it belong in a separate package? (Currently the script relies on living in the directory with the files it is running against). @elastic/kibana-operations 
- Is the current directory the right location for the generated docs? Due to the way docs are released, we need the generated docs to be committed to the repo. @elastic/kibana-operations 
- Are the first set of generated response ops docs acceptable? @kobelb 
- What would the level of effort be to, as a follow-up task, automate the running of this script in CI and auto-commiting the output on PRs? @elastic/kibana-operations (for now docs team has agreed to handle this manually)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/245287)
* Commit: 5efed752419df94184c890e89145975d6a0fe44b

### Failed CI Steps
* [Jest Integration Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bac5-ae72-4545-968e-bbfc6d777e03)
* [Jest Integration Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192baea-718b-4d12-9cbb-281b5c9f1c6a)
* [FTR Configs #18](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bacb-0e14-46ff-9273-da3cfac3f08d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bac5-ae72-4545-968e-bbfc6d777e03) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192bac5-ae72-4545-968e-bbfc6d777e03/artifacts/0192baea-3dd8-4431-ba47-1d21f5246c00) Jest Integration Tests #<span></span>2 / incompatible_cluster_routing_allocation retries the INIT action with a descriptive message when cluster settings are incompatible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192baea-718b-4d12-9cbb-281b5c9f1c6a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192baea-718b-4d12-9cbb-281b5c9f1c6a/artifacts/0192bafd-da16-417f-b6a2-79b61e94ba10) Jest Integration Tests #<span></span>2 / incompatible_cluster_routing_allocation retries the INIT action with a descriptive message when cluster settings are incompatible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bac5-ae72-4545-968e-bbfc6d777e03) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192bac5-ae72-4545-968e-bbfc6d777e03/artifacts/0192baea-3dd4-4aa5-b263-3e2009ae06d2) Jest Integration Tests #<span></span>2 / migration v2 - read batch size does not reduce the read batchSize in half if no batches exceeded maxReadBatchSizeBytes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192baea-718b-4d12-9cbb-281b5c9f1c6a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192baea-718b-4d12-9cbb-281b5c9f1c6a/artifacts/0192bafd-da15-46fe-b031-1edc87f30142) Jest Integration Tests #<span></span>2 / migration v2 - read batch size does not reduce the read batchSize in half if no batches exceeded maxReadBatchSizeBytes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bac5-ae72-4545-968e-bbfc6d777e03) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192bac5-ae72-4545-968e-bbfc6d777e03/artifacts/0192baea-3dd7-49b7-a5b9-f0800160f0af) Jest Integration Tests #<span></span>2 / migration v2 - read batch size reduces the read batchSize in half if a batch exceeds maxReadBatchSizeBytes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192baea-718b-4d12-9cbb-281b5c9f1c6a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192baea-718b-4d12-9cbb-281b5c9f1c6a/artifacts/0192bafd-da15-46ba-9ac9-6f4120ce05ef) Jest Integration Tests #<span></span>2 / migration v2 - read batch size reduces the read batchSize in half if a batch exceeds maxReadBatchSizeBytes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192bac5-ae72-4545-968e-bbfc6d777e03) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192bac5-ae72-4545-968e-bbfc6d777e03/artifacts/0192baea-3dd7-4c72-8c3b-d8e71b48917d) Jest Integration Tests #<span></span>2 / migration v2 migrates saved objects normally with multiple ES nodes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/245287#0192baea-718b-4d12-9cbb-281b5c9f1c6a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/245287/jobs/0192baea-718b-4d12-9cbb-281b5c9f1c6a/artifacts/0192bafd-da12-459d-9d03-48f30b246256) Jest Integration Tests #<span></span>2 / migration v2 migrates saved objects normally with multiple ES nodes

### History
* :broken_heart: [Build #241218](https://buildkite.com/elastic/kibana-pull-request/builds/241218) failed df17b448c949ea11b8207607de57a62a165c5efc
* :broken_heart: [Build #241164](https://buildkite.com/elastic/kibana-pull-request/builds/241164) failed 5c31ac4b69242527767355b204c36dca57471f79

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245287","number":245287,"commit":"5efed752419df94184c890e89145975d6a0fe44b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241218","number":241218,"commit":"df17b448c949ea11b8207607de57a62a165c5efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241164","number":241164,"commit":"5c31ac4b69242527767355b204c36dca57471f79"}],"number":245287}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/245288)
* Commit: c9bb65e814dba9faad8b1612647b35f024d95ee4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #245287](https://buildkite.com/elastic/kibana-pull-request/builds/245287) failed 5efed752419df94184c890e89145975d6a0fe44b
* :broken_heart: [Build #241218](https://buildkite.com/elastic/kibana-pull-request/builds/241218) failed df17b448c949ea11b8207607de57a62a165c5efc
* :broken_heart: [Build #241164](https://buildkite.com/elastic/kibana-pull-request/builds/241164) failed 5c31ac4b69242527767355b204c36dca57471f79

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245288","number":245288,"commit":"c9bb65e814dba9faad8b1612647b35f024d95ee4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245287","number":245287,"commit":"5efed752419df94184c890e89145975d6a0fe44b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241218","number":241218,"commit":"df17b448c949ea11b8207607de57a62a165c5efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241164","number":241164,"commit":"5c31ac4b69242527767355b204c36dca57471f79"}],"number":245288}
buildkite-pr-comment-->

kilfoyle commented:
@elastic/kibana-operations, @kobelb  Apologies for the ping but I'd really appreciate if someone can help with this PR and Luke's questions [above](https://github.com/elastic/kibana/pull/191787/#pullrequestreview-2375968805).

jbudz commented:
> Most important thing to review: are we generally comfortable with the yml schema design here? Because this will dictate the authoring experience for yml config docs in the future

Seems fine, it covers all the needs of existing settings and we'll be in a position to adjust if needed

> Is the current directory the right location for the conversion script or does it belong in a separate package? (Currently the script relies on living in the directory with the files it is running against).

We don't run CI on changes only in the docs folder but if validation is left to Kibana CI we'll want to make an adjustment to https://github.com/elastic/kibana/blob/main/.buildkite/pull_requests.json#L24. 

If it remains static I don't see issues.  If not,  it could be difficult to maintain across all repositories with docs and all branches.  It does need to be accessible locally by contributors in some form and this is easiest in that regard.

> Is the current directory the right location for the generated docs? Due to the way docs are released, we need the generated docs to be committed to the repo

:+1: 

> What would the level of effort be to, as a follow-up task, automate the running of this script in CI and auto-commiting the output on PRs? 

Low, except we shut off Kibana CI on old branches.  We would have to a create a docs only check, similar to 
buildkite/docs-build-pr.  Preferably the check could be baked into that one.

kobelb commented on docs/settings-gen/source/kibana-alert-action-settings.asciidoc:
```diff
@@ -0,0 +1,1033 @@
+====
+Specifies HTTP headers for the proxy, if using a proxy for actions.
+
+Default: `HASH(0x12bb51a88)` +
```
This should be the following:

```
Default: {}
```

But it renders as the following:

<img width="233" alt="Screenshot 2024-11-08 at 2 49 38 PM" src="https://github.com/user-attachments/assets/aaa09330-a299-4b87-b992-7b24f12b960a">

kilfoyle replied:
Another nice catch. Fixed in [fa1fe32](https://github.com/elastic/kibana/pull/191787/commits/fa1fe3211eecf3168dacd48bce888eb500c7537d)

kobelb commented on docs/settings-gen/source/kibana-alert-action-settings.asciidoc:
```diff
@@ -0,0 +1,1033 @@
+* `put`
+* `patch`
+
+Default: `patch` +
```
The default is `put`.

kilfoyle replied:
Thanks for the nice catch! Fixed in [fa1fe32](https://github.com/elastic/kibana/pull/191787/commits/fa1fe3211eecf3168dacd48bce888eb500c7537d)

kobelb reviewed (commented):
Two small corrections.

The following is just my opinion, and can be ignored:

1. Having the example in a separate file is cumbersome. Is it possible for these to be inline and use the multi-line YAML strings?
2. Rendering each setting in a collapsible section made consuming these docs hard. However, I'm quite adjusted to the old format, so take this with a grain of salt.

kilfoyle commented:
@kobelb, thanks very much for the excellent suggestions!

> Two small corrections.

Both are fixed.

> Having the example in a separate file is cumbersome. Is it possible for these to be inline and use the multi-line YAML strings?

Indeed, it's a lot less cumbersome having the examples included in the YAML rather than in a separate file, so I've made that change.

> Rendering each setting in a collapsible section made consuming these docs hard. However, I'm quite adjusted to the old format, so take this with a grain of salt.

I was thinking that the collapsible sections looked tidier, but on further thought I agree that the usability is much better without them.

If you have other suggestions to improve the formatting I'm happy to add them in.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/254018)
* Commit: fd7d0e2518ac07c56ce50a0d6270cdbe19035718

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73
* :green_heart: [Build #245288](https://buildkite.com/elastic/kibana-pull-request/builds/245288) succeeded c9bb65e814dba9faad8b1612647b35f024d95ee4
* :broken_heart: [Build #245287](https://buildkite.com/elastic/kibana-pull-request/builds/245287) failed 5efed752419df94184c890e89145975d6a0fe44b
* :broken_heart: [Build #241218](https://buildkite.com/elastic/kibana-pull-request/builds/241218) failed df17b448c949ea11b8207607de57a62a165c5efc
* :broken_heart: [Build #241164](https://buildkite.com/elastic/kibana-pull-request/builds/241164) failed 5c31ac4b69242527767355b204c36dca57471f79

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246369","number":246369,"commit":"ae43bf4247c349e6801fab348e65464357de4b73"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245288","number":245288,"commit":"c9bb65e814dba9faad8b1612647b35f024d95ee4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245287","number":245287,"commit":"5efed752419df94184c890e89145975d6a0fe44b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241218","number":241218,"commit":"df17b448c949ea11b8207607de57a62a165c5efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241164","number":241164,"commit":"5c31ac4b69242527767355b204c36dca57471f79"}],"number":254018}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/256673)
* Commit: 035e4598d4f7f11020a244c9c0a8e799494792a6

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #254018](https://buildkite.com/elastic/kibana-pull-request/builds/254018) succeeded fd7d0e2518ac07c56ce50a0d6270cdbe19035718
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73
* :green_heart: [Build #245288](https://buildkite.com/elastic/kibana-pull-request/builds/245288) succeeded c9bb65e814dba9faad8b1612647b35f024d95ee4
* :broken_heart: [Build #245287](https://buildkite.com/elastic/kibana-pull-request/builds/245287) failed 5efed752419df94184c890e89145975d6a0fe44b
* :broken_heart: [Build #241218](https://buildkite.com/elastic/kibana-pull-request/builds/241218) failed df17b448c949ea11b8207607de57a62a165c5efc
* :broken_heart: [Build #241164](https://buildkite.com/elastic/kibana-pull-request/builds/241164) failed 5c31ac4b69242527767355b204c36dca57471f79

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256673","number":256673,"commit":"035e4598d4f7f11020a244c9c0a8e799494792a6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246369","number":246369,"commit":"ae43bf4247c349e6801fab348e65464357de4b73"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245288","number":245288,"commit":"c9bb65e814dba9faad8b1612647b35f024d95ee4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245287","number":245287,"commit":"5efed752419df94184c890e89145975d6a0fe44b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241218","number":241218,"commit":"df17b448c949ea11b8207607de57a62a165c5efc"}],"number":256673}
buildkite-pr-comment-->

kilfoyle commented:
@jbudz and @lukeelmers thanks a lot for your feedback on this PR!

I've addressed the suggestions so far. Are there any changes you'd like or do you think this is ready to merge?

lukeelmers commented:
No concerns on my end as long as we get a ✅ from @jbudz or someone on the ops team.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/260073)
* Commit: 72135bcc8e23ec23dccf31d6347737be29ffc479

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #256673](https://buildkite.com/elastic/kibana-pull-request/builds/256673) succeeded 035e4598d4f7f11020a244c9c0a8e799494792a6
* :green_heart: [Build #254018](https://buildkite.com/elastic/kibana-pull-request/builds/254018) succeeded fd7d0e2518ac07c56ce50a0d6270cdbe19035718
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73
* :green_heart: [Build #245288](https://buildkite.com/elastic/kibana-pull-request/builds/245288) succeeded c9bb65e814dba9faad8b1612647b35f024d95ee4
* :broken_heart: [Build #245287](https://buildkite.com/elastic/kibana-pull-request/builds/245287) failed 5efed752419df94184c890e89145975d6a0fe44b
* :broken_heart: [Build #241218](https://buildkite.com/elastic/kibana-pull-request/builds/241218) failed df17b448c949ea11b8207607de57a62a165c5efc

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260073","number":260073,"commit":"72135bcc8e23ec23dccf31d6347737be29ffc479"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256673","number":256673,"commit":"035e4598d4f7f11020a244c9c0a8e799494792a6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246369","number":246369,"commit":"ae43bf4247c349e6801fab348e65464357de4b73"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245288","number":245288,"commit":"c9bb65e814dba9faad8b1612647b35f024d95ee4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245287","number":245287,"commit":"5efed752419df94184c890e89145975d6a0fe44b"}],"number":260073}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/260879)
* Commit: 3008521b4ed2154312bd528e5eb38af4637eac63

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #260073](https://buildkite.com/elastic/kibana-pull-request/builds/260073) succeeded 72135bcc8e23ec23dccf31d6347737be29ffc479
* :green_heart: [Build #256673](https://buildkite.com/elastic/kibana-pull-request/builds/256673) succeeded 035e4598d4f7f11020a244c9c0a8e799494792a6
* :green_heart: [Build #254018](https://buildkite.com/elastic/kibana-pull-request/builds/254018) succeeded fd7d0e2518ac07c56ce50a0d6270cdbe19035718
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73
* :green_heart: [Build #245288](https://buildkite.com/elastic/kibana-pull-request/builds/245288) succeeded c9bb65e814dba9faad8b1612647b35f024d95ee4
* :broken_heart: [Build #245287](https://buildkite.com/elastic/kibana-pull-request/builds/245287) failed 5efed752419df94184c890e89145975d6a0fe44b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260879","number":260879,"commit":"3008521b4ed2154312bd528e5eb38af4637eac63"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260073","number":260073,"commit":"72135bcc8e23ec23dccf31d6347737be29ffc479"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256673","number":256673,"commit":"035e4598d4f7f11020a244c9c0a8e799494792a6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246369","number":246369,"commit":"ae43bf4247c349e6801fab348e65464357de4b73"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/245288","number":245288,"commit":"c9bb65e814dba9faad8b1612647b35f024d95ee4"}],"number":260879}
buildkite-pr-comment-->

kilfoyle commented:
Based on feedback requests I've made a few changes: 
 - Each setting description is now a whitespace-preserving multiline string, rather than an array of strings (it's much cleaner this way).
 - Each setting now has a mandatory "datatype" field that can be one of `string`/`bool`/`int`/`float`/`enum`. When `enum` is selected, the list of supported options should be defined using the `options` keys.
 - Tidied up the readme.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/261610)
* Commit: a4cbb70a2debfa0bf9764f072c02d0813b5ede48

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #260879](https://buildkite.com/elastic/kibana-pull-request/builds/260879) succeeded 3008521b4ed2154312bd528e5eb38af4637eac63
* :green_heart: [Build #260073](https://buildkite.com/elastic/kibana-pull-request/builds/260073) succeeded 72135bcc8e23ec23dccf31d6347737be29ffc479
* :green_heart: [Build #256673](https://buildkite.com/elastic/kibana-pull-request/builds/256673) succeeded 035e4598d4f7f11020a244c9c0a8e799494792a6
* :green_heart: [Build #254018](https://buildkite.com/elastic/kibana-pull-request/builds/254018) succeeded fd7d0e2518ac07c56ce50a0d6270cdbe19035718
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73
* :green_heart: [Build #245288](https://buildkite.com/elastic/kibana-pull-request/builds/245288) succeeded c9bb65e814dba9faad8b1612647b35f024d95ee4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261610","number":261610,"commit":"a4cbb70a2debfa0bf9764f072c02d0813b5ede48"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260879","number":260879,"commit":"3008521b4ed2154312bd528e5eb38af4637eac63"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260073","number":260073,"commit":"72135bcc8e23ec23dccf31d6347737be29ffc479"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256673","number":256673,"commit":"035e4598d4f7f11020a244c9c0a8e799494792a6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246369","number":246369,"commit":"ae43bf4247c349e6801fab348e65464357de4b73"}],"number":261610}
buildkite-pr-comment-->

kilfoyle commented:
There's an error to do with the codeowners file that I'm not sure how to fix:

```
Unknown owner on line 3325: make sure the team @elastic/obs-ux-onboarding-team exists, is publicly visible, and has write access to the repository
```

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262087)
* Commit: 519cdf94da82728a7b81b2227ef1ae621044d049

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #261610](https://buildkite.com/elastic/kibana-pull-request/builds/261610) succeeded a4cbb70a2debfa0bf9764f072c02d0813b5ede48
* :green_heart: [Build #260879](https://buildkite.com/elastic/kibana-pull-request/builds/260879) succeeded 3008521b4ed2154312bd528e5eb38af4637eac63
* :green_heart: [Build #260073](https://buildkite.com/elastic/kibana-pull-request/builds/260073) succeeded 72135bcc8e23ec23dccf31d6347737be29ffc479
* :green_heart: [Build #256673](https://buildkite.com/elastic/kibana-pull-request/builds/256673) succeeded 035e4598d4f7f11020a244c9c0a8e799494792a6
* :green_heart: [Build #254018](https://buildkite.com/elastic/kibana-pull-request/builds/254018) succeeded fd7d0e2518ac07c56ce50a0d6270cdbe19035718
* :green_heart: [Build #246369](https://buildkite.com/elastic/kibana-pull-request/builds/246369) succeeded ae43bf4247c349e6801fab348e65464357de4b73

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262087","number":262087,"commit":"519cdf94da82728a7b81b2227ef1ae621044d049"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261610","number":261610,"commit":"a4cbb70a2debfa0bf9764f072c02d0813b5ede48"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260879","number":260879,"commit":"3008521b4ed2154312bd528e5eb38af4637eac63"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260073","number":260073,"commit":"72135bcc8e23ec23dccf31d6347737be29ffc479"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/256673","number":256673,"commit":"035e4598d4f7f11020a244c9c0a8e799494792a6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/254018","number":254018,"commit":"fd7d0e2518ac07c56ce50a0d6270cdbe19035718"}],"number":262087}
buildkite-pr-comment-->

jbudz commented:
Where did this land on automation?  Are we leaving it as a manual process for now?

kilfoyle commented:
> Where did this land on automation? Are we leaving it as a manual process for now?

@jbudz Thanks to [Add initial support for structured settings references (yaml](https://github.com/elastic/docs-builder/pull/105)), in the upcoming new Docs V3, processing of the yaml settings files will be handled automatically as part of the docs build (@Mpdreamz please correct me if that's not right).

For sure, the Perl script added in this PR will not need to be run automatically, and it will exist just temporarily until we arrive at the new and better docs system.
